src/main/java/com/example/W5Assessment/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,83 @@
+package com.example.W5Assessment.exception;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.LocalDateTime;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    private static final Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
+
+    @ExceptionHandler(value = NoSuchTaskExistsException.class)
+    public ResponseEntity<Object> noSuchPetExistsException(NoSuchTaskExistsException ex, WebRequest request) {
+        // Log the exception
+        logException(ex);
+
+        // Return a response entity with an error message
+        ErrorMessage errorMessage = new ErrorMessage(
+                HttpStatus.NOT_FOUND,
+                ex.getMessage(),
+                LocalDateTime.now());
+        return new ResponseEntity<>(errorMessage, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(value = EmptyTaskListException.class)
+    public ResponseEntity<Object> EmptyTaskListException(EmptyTaskListException ex, WebRequest request) {
+        // Log the exception
+        logException(ex);
+
+        // Return a response entity with an error message
+        ErrorMessage errorMessage = new ErrorMessage(
+                HttpStatus.NOT_FOUND,
+                ex.getMessage(),
+                LocalDateTime.now());
+        return new ResponseEntity<>(errorMessage, HttpStatus.NOT_FOUND);
+    }
+
+    // Method to log the exception to a text file
+    private void logException(Exception ex) {
+        try (FileWriter writer = new FileWriter("error.log", true);
```

Comment:
you should use a logging library to do the logging
Log back or slf4j or log4j or sprinboot logging etc etc. They automaticallt log to log files when you do log.info() or log.error()